rasa/shared/nlu/training_data/training_data.py

```diff
@@ -679,38 +678,54 @@ def can_train_nlu_model(self) -> bool:
         ]
         return not any([len(lst) > 0 for lst in lists_to_check])
 
-    def divide_into_chunks(self, max_size: int = 2) -> List["TrainingDataChunk"]:
+    def divide_into_chunks(self, num_chunks: int) -> List["TrainingDataChunk"]:
         """Divides the training data into smaller chunks.
 
         Each chunk should be a good representation of the complete dataset. E.g. it
         should not just contain examples of one intent, but instead match the
         distribution of the complete dataset.
 
         Args:
-            max_size: The maximum size (in MB) of one chunk.
+            num_chunks: The total number of chunks into which the training data should be broken.
 
         Returns:
             A list of all training data chunks.
         """
-        pass
+        all_chunks = []
+        data_to_chunk = copy.deepcopy(self)
 
+        for chunk_index in range(num_chunks):
 
-class TrainingDataChunk:
-    """Holds a portion of the complete TrainingData."""
+            chunk_size_fraction = 1 / (num_chunks - chunk_index)
+            current_chunk, leftover_examples = data_to_chunk.split_nlu_examples(
+                1 - chunk_size_fraction
+            )
 
-    def __init__(
-        self,
-        training_examples: List[Message],
-        responses: Optional[Dict[Text, List[Dict[Text, Any]]]] = None,
-    ) -> None:
-        """Initialize a training data chunk.
+            if chunk_index == num_chunks - 1:
+                # There can be a small number of leftover examples
+                # during the last bit of chunking because of rounding errors.
+                current_chunk.extend(leftover_examples)
+
+            # update the data to chunk in next iteration
+            data_to_chunk = TrainingData(
+                leftover_examples,
+                responses=data_to_chunk._needed_responses_for_examples(
+                    leftover_examples
+                ),
+            )
+            all_chunks.append(
+                TrainingDataChunk(
+                    current_chunk,
+                    responses=data_to_chunk._needed_responses_for_examples(
+                        current_chunk
+                    ),
+                )
+            )
+        return all_chunks
 
-        Args:
-            training_examples: List of messages representing the training data.
-            responses: List of responses.
-        """
-        self.training_examples = training_examples
-        self.responses = responses
+
+class TrainingDataChunk(TrainingData):
```

Comment:
Can we overwrite the `__init__` method to make sure this object has no regexes, lookup tables, etc. set?

Reply:
Would throwing a warning if they are set enough?

Reply:
I am still not super clear on what's the downside if `TrainingDataChunk` have regexes, lookup tables, etc. Is it that methods like `persist_chunk` and `load_chunk` will need to factor them in as well? If yes then sub-classing also doesn't make sense because it's very unusual for a sub-class to not have attributes of the parent class. What do you think? 😅

Reply:
Is more about the memory used by this object. Lookup tables etc. can also take up quite some memory and we don't need them anymore at the point where we split the training data into smaller chunks. So we don't want to store them anymore. I am just afraid, that if we keep the parameters, that people think that they can use them, which we don't want.

Reply:
Hmm, okay. Then I'll raise an exception in the overloaded constructor if they are set by the parent constructor. Does that sound good?

Reply:
Added an exception before the parent constructor is even called.